app/src/test/java/com/fstyle/structure_android/data/source/UserRepositoryTest.java

```diff
@@ -0,0 +1,104 @@
+package com.fstyle.structure_android.data.source;
+
+import com.fstyle.structure_android.data.model.User;
+import com.fstyle.structure_android.data.model.UsersList;
+import com.fstyle.structure_android.data.source.remote.UserRemoteDataSource;
+import com.fstyle.structure_android.data.source.remote.api.service.NameApi;
+import java.util.ArrayList;
+import java.util.List;
+import okhttp3.MediaType;
+import okhttp3.ResponseBody;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import retrofit2.Response;
+import retrofit2.adapter.rxjava.HttpException;
+import rx.Observable;
+import rx.observers.TestSubscriber;
+
+/**
+ * Created by Sun on 3/11/2017.
+ */
+public class UserRepositoryTest {
+
+    private static final String USER_LOGIN_1 = "abc";
+    private static final String USER_LOGIN_2 = "def";
+
+    @Mock
+    NameApi mNameApi;
+
+    private UserRepository mUserRepository;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        mUserRepository = new UserRepository(null, new UserRemoteDataSource(mNameApi));
+    }
+
+    @After
+    public void tearDown() throws Exception {
+
+    }
+
+    @Test
+    public void searchUsers_200ResponseCode_InvokesCorrectApiCalls() {
```

Comment:
ERROR: [MethodNamingConventions] Method names should not contain underscores
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#MethodNamingConventions